server/interpreter/data/object.go

```diff
@@ -16,11 +16,48 @@
 
 package data
 
-// Object represents typical JavaScript objects with (optional)
-// prototype, properties, etc.
-type Object struct {
+// Object represents any JavaScript value (primitive, object, etc.).
+type Object interface {
+	/// Any Object is a valid Value.
+	Value
+
+	// Proto returns the prototype (parent) object for this object.
+	// N.B. this is object.__proto__, not Constructor.prototype!
+	Proto() Object
+
+	// Get returns the current value of the given property or an
+	// ErrorMsg if that was not possible.
+	Get(name string) (Value, *ErrorMsg)
```

Comment:
The intention is that ErrorMsg objects returned by the data package will be turned into a new JS `Error` object of the appropriate type by the caller, which would then be thrown.  However:

- I haven't actually written the code to do that yet.
- It might be better to replace `Name string` with an enum-valued field, since there are only about a half-dozen different error types (`SyntaxError`, `TypeError`, `ReferenceError` etc.)
- It would be nice to replace the error return with a "throw" flag and just create and return the relevant JS `Error` object as the first return value, but that requires a prototype and I want to avoid having to pass a Protos object to every single function/method that might need to throw an error.
- Fulfilling the Go `error` interface was mainly so that I could `panic()` these errors as a temporary measure until I implement the code to cause them to be turned into `Error`s and and throw.

I would be happy to prepare an improved version of this as part of this PR, but I invite your advice before I do so.